### PR TITLE
test: got the client_integration_test moved over to the builder.

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low
Testing: yep
Release Notes: inline
Fixes https://github.com/envoyproxy/envoy-mobile/issues/2303

### CODE REVIEW

**Comment by alyssawilk** [comment]
@goaway I expect this to fail CI as I haven't gotten the RTDS test passing yet, but client integration test is good to go and there's a lot of moving parts here I'd like your thoughts on, especially the weirdness of taking config from the builder, using it to initialize the integration test config helper, then serializing the config helper back to the builder so we can test custom configs.
(not sure how else we do RTDS, happy to have your thoughts here as I can tweak in parallel with fixing up the rtds test)

thanks!

**Comment by goaway** [comment]
Discussed a bit over Slack. Overall this seems fine to me, though at least on a case-by-case basis I think it's worth considering whether a given override might simply make sense as a Builder option. I'll also be thinking over possibilities for allowing for direct overrides - at least on C++ - rather than needing to do the serialization ahead of time and passing that back through.

File: library/cc/stream.h [hunk 1]
```diff
@@ -18,7 +18,7 @@ class Stream {
   Stream(envoy_engine_t engine_handle, envoy_stream_t handle);
 
   Stream& sendHeaders(RequestHeadersSharedPtr headers, bool end_stream);
-  Stream& sendData(envoy_data data);
+  Stream& sendData(envoy_data data, bool end_stream);
```
[the remark this answers is not in the document]
Comment (alyssawilk) [reply library/cc/stream.h:21]: but doesn't sendHeaders with end stream implicitly close?  That's a pretty weird abstraction IMO but yeah this PR is not the place to change it.

**Comment by alyssawilk** [comment]
ok as discussed, moved the default to "use builder config" but I don't think we want to block this PR on builder RTDS support.

File: library/cc/engine_builder.h [hunk 2]
```diff
@@ -46,6 +47,9 @@ class EngineBuilder {
   // Filter): EngineBuilder& addNativeFilter(name: String = UUID.randomUUID().toString(),
   // typedConfig: String): EngineBuilder& addStringAccessor(name: String, accessor:
   // EnvoyStringAccessor): EngineBuilder {
+protected:
+  void setOverrideConfigForTests(std::string config) { config_override_for_tests_ = config; }
+  void setAdminAddressPathForTests(std::string admin) { admin_address_path_for_tests_ = admin; }
```
Comment (alyssawilk) [inline library/cc/engine_builder.h:52]: alternately could publicly add key-value and hash map for generic command line override.

File: library/cc/request_headers_builder.cc [hunk 3]
```diff
@@ -28,6 +28,11 @@ RequestHeadersBuilder::addUpstreamHttpProtocol(UpstreamHttpProtocol upstream_htt
   return *this;
 }
 
+RequestHeadersBuilder& RequestHeadersBuilder::addKeyValue(std::string key, std::string value) {
```
Comment (goaway) [inline library/cc/request_headers_builder.cc:31]: nit: I'd call this`set`, for alignment with the other builders: https://github.com/envoyproxy/envoy-mobile/blob/main/library/kotlin/io/envoyproxy/envoymobile/RequestHeadersBuilder.kt#L43

**Review by goaway — DISMISSED** [review]

**Review by jpsim — DISMISSED** [review]
Yay, one more up to date API!

**Pull request merged**